tests/testthat/test-mock_httr_response.R

```diff
@@ -0,0 +1,22 @@
+context('mock httr response function')
+
+test_that("Mocking works with simple JSON example", {
+  out <-   getFromNamespace("response", "httr")(
```

Comment:
Usually it's a bad idea to replicate code from the function you're implementing in an actual test, since you end up implementing it twice making it more cumbersome to make changes in the future. A different way is to test properties of the output:

```r
expect_equal(result$status_code, 200L)
expect_equal(result$content, jsonlite::toJSON(list(data = "it works!"), auto_unbox = TRUE))
expect_equal(result$headers$`Content-Type`, "application/json")
expect_is(result, "response")
```